### ISSUE TITLE
Cannot build with "invalid Monero source dir"

### ISSUE THREAD

**Issue by sethsimmons** [vtnerd/monero-lws#17]
Hi there,

Trying to get this built but running into an issue. When I attempt to build mymonero-lws after building Monero froms ource, I keep getting this error message:

```
➜  build git:(develop) cmake -DMONERO_SOURCE_DIR=~/build/monero -DMONERO_BUILD_DIR=~/build/monero/build ..                                                             
-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is GNU 9.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:89 (load_cache):
  load_cache Cannot load cache file from ~/build/monero/build/CMakeCache.txt


CMake Error at CMakeLists.txt:105 (message):
  Invalid Monero source dir - does not appear to match source used for build
  directory


-- Configuring incomplete, errors occurred!
See also "/home/tr/build/monero-lws/build/CMakeFiles/CMakeOutput.log".
```

Monero is cloned in `~/build/monero` and built against `release-v0.17`, with the resulting build files in `~/build/monero/build/Linux/release-v0.17/release`. I have tried passing both `~/build/monero/build` and `~/build/monero/build/Linux/release-v0.17/release` for the `-DMONERO_BUILD_DIR` flag, but neither work.

**Comment by sethsimmons** [comment]
Contents of each directory that should be the correct parameters is below:

```
➜  build git:(develop) ✗ ls -la ~/build/monero
total 324
drwxrwxr-x 13 tr tr   4096 Mar  1 15:12 .
drwxrwxr-x 12 tr tr   4096 Mar  1 15:00 ..
drwxrwxr-x  9 tr tr   4096 Mar  1 15:15 .git
-rw-rw-r--  1 tr tr     47 Mar  1 15:00 .gitattributes
drwxrwxr-x  3 tr tr   4096 Mar  1 15:00 .github
-rw-rw-r--  1 tr tr   1085 Mar  1 15:12 .gitignore
-rw-rw-r--  1 tr tr    693 Mar  1 15:12 .gitmodules
-rw-rw-r--  1 tr tr   3815 Mar  1 15:12 .travis.yml
-rw-rw-r--  1 tr tr  10083 Mar  1 15:12 ANONYMITY_NETWORKS.md
-rw-rw-r--  1 tr tr  39025 Mar  1 15:12 CMakeLists.txt
-rw-rw-r--  1 tr tr   7331 Mar  1 15:00 CMakeLists_IOS.txt
-rw-rw-r--  1 tr tr  11874 Mar  1 15:12 CONTRIBUTING.md
-rw-rw-r--  1 tr tr   6683 Mar  1 15:12 Dockerfile
-rw-rw-r--  1 tr tr 100741 Mar  1 15:12 Doxyfile
-rw-rw-r--  1 tr tr   7054 Mar  1 15:12 LEVIN_PROTOCOL.md
-rw-rw-r--  1 tr tr   2730 Mar  1 15:12 LICENSE
-rw-rw-r--  1 tr tr   9546 Mar  1 15:00 Makefile
-rw-rw-r--  1 tr tr   2471 Mar  1 15:12 README.i18n.md
-rw-rw-r--  1 tr tr  44215 Mar  1 15:12 README.md
-rw-rw-r--  1 tr tr   3067 Mar  1 15:12 ZMQ.md
drwxrwxr-x  3 tr tr   4096 Mar  1 15:01 build
drwxrwxr-x  2 tr tr   4096 Mar  1 15:12 cmake
drwxrwxr-x 10 tr tr   4096 Mar  1 15:12 contrib
drwxrwxr-x 12 tr tr   4096 Mar  1 15:12 external
drwxrwxr-x  2 tr tr   4096 Mar  1 15:00 include
drwxrwxr-x 30 tr tr   4096 Mar  1 15:12 src
drwxrwxr-x 18 tr tr   4096 Mar  1 15:12 tests
drwxrwxr-x  2 tr tr   4096 Mar  1 15:00 translations
drwxrwxr-x 10 tr tr   4096 Mar  1 15:12 utils
➜  build git:(develop) ✗ ls -la ~/build/monero/build
total 12
drwxrwxr-x  3 tr tr 4096 Mar  1 15:01 .
drwxrwxr-x 13 tr tr 4096 Mar  1 15:12 ..
drwxrwxr-x  4 tr tr 4096 Mar  1 15:15 Linux
```

**Comment by sethsimmons** [comment]
CMakeCache.txt file is only available in `~/build/monero/build/Linux/release-v0.17/release/`:

```
➜  build git:(develop) ✗ ls -la ~/build/monero/build/CMakeCache.txt
ls: cannot access '/home/tr/build/monero/build/CMakeCache.txt': No such file or directory
➜  build git:(develop) ✗ ls -la ~/build/monero/build/Linux/release-v0.17/release/CMakeCache.txt
-rw-rw-r-- 1 tr tr 52142 Mar  1 15:15 /home/tr/build/monero/build/Linux/release-v0.17/release/CMakeCache.txt
```

**Comment by sethsimmons** [comment]
Switching from ~ to a literal path for both source and build directories worked.

Functioning commands were:

```
cmake -DMONERO_SOURCE_DIR=/home/tr/monero -DMONERO_BUILD_DIR=/home/tr/monero/build/Linux/release-v0.17/release .. 
```
